ddl/delete_range.go

```diff
@@ -235,6 +237,22 @@ func (dr *delRange) doTask(ctx sessionctx.Context, r util.DelRangeTask) error {
 				return errors.Trace(err)
 			}
 			startKey, endKey := r.Range()
+
+			tableID := tablecodec.DecodeTableID(endKey)
+			tableID -= 1
+
+			tikvStore, ok := dr.store.(helper.Storage)
+			if ok {
+				tikvHelper := &helper.Helper{
+					Store:       tikvStore,
+					RegionCache: tikvStore.GetRegionCache(),
+				}
+				ruleID := fmt.Sprintf("table-%v-r", tableID)
+				// If DeletePlacementRule fails here, the rule will be deleted in `HandlePlacementRuleRoutine`.
```

Comment:
This only happens when mocking

Reply:
This is incredibly ugly. I suggest removing it. We can handle all removing of rules in `doGCplacement`.